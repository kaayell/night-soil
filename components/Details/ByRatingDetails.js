import React from 'react'
import {View} from "react-native";
import RatingPieChart from "../Charts/RatingPieChart";

const ByRatingDetails = (props) => {
  console.log(props)
  return (
    <View>
    <RatingPieChart/>
    </View>
  )
}

export default ByRatingDetails