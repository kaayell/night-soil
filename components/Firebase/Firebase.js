import * as firebase from 'firebase'
import { ANDROID_API_KEY, config, IOS_API_KEY } from './secrets'

export default class Firebase {
  static init () {
    firebase.initializeApp(config)
  }

  static getAuth() {
    return firebase.auth()
  }

  static getUserDetailsRef() {
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/user/details/${uid}`)
  }

  static savePoop(poopDetails) {
    let currentUser = firebase.auth().currentUser
    firebase.database().ref(`/poop/${currentUser.uid}`).push().set(poopDetails)
  }

  static getPoopsRef(){
    let uid = firebase.auth().currentUser.uid
    return firebase.database().ref(`/poop/${uid}`)
  }

  static async loginWithGoogle () {
    const response = await Expo.Google.logInAsync({
      androidClientId: ANDROID_API_KEY,
      iosClientId: IOS_API_KEY,
      scopes: ['profile', 'email'],
    })

    if (response.type === 'success') {
      const credential = firebase.auth.GoogleAuthProvider.credential(response.idToken, response.accessToken)
      firebase.auth().signInWithCredential(credential)
    }
  }
}