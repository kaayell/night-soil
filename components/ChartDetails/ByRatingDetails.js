import React, {Component} from 'react'
import {View, StatusBar, TouchableOpacity, Platform, Text} from "react-native";
import {PieChart} from "react-native-svg-charts";
import Firebase from "../Firebase/Firebase";
import {pieData, RATING_COLORS} from "../ChartHelpers/pie_chart_helper";
import {Icon} from "react-native-elements";
import {BLUE, OFF_WHITE} from "../StyleGuide/colors";
import _ from "lodash";
import {POPPINS} from "../StyleGuide/fonts";
import styles from "../StyleGuide/styles"

class ByRatingDetails extends Component {
  constructor(props) {
    super(props)

    this.state = {
      poopData: [],
      selectedSlice: null
    }
  }

  componentDidMount() {
    Firebase.getPoopsRef().on('value', snapshot => {
      let poopData = []
      snapshot.forEach(childSnapshot => {
        const poop = childSnapshot.val()
        poopData.push(poop)
      })
      this.setState({poopData})
    })
  }

  render() {
    let grouped = _.groupBy(this.state.poopData, 'poopRating')
    let pieData = _.map(grouped, (groupValue, groupKey) => {
      return {
        value: groupValue.length,
        svg: {
          fill: RATING_COLORS[groupKey]
        },
        key: `${Math.random()}`,
        onPress: () => this.setState({
          selectedSlice: {
            key: groupKey,
            value: groupValue.length,
            color: RATING_COLORS[groupKey]
          }
        })
      }
    })

    const paddingTop = Platform.OS === 'ios' ? (parseInt(Platform.Version, 10) > 10 ? 35 : 20) : StatusBar.currentHeight + 10
    return (
      <View style={{flex: 1, backgroundColor: OFF_WHITE, paddingTop, height: "100%"}}>
        <TouchableOpacity style={{flexDirection: 'row', height: '5%', justifyContent: 'flex-end', paddingRight: 15}}
                          onPress={() => this.props.navigation.goBack()}>
          <Icon name={'close'} color={BLUE}/>
        </TouchableOpacity>
        <View style={{flex: 1}}>
          <PieChart
            style={{height: 300}}
            data={pieData}
            spacing={0}
            innerRadius={10}
            outerRadius={'95%'}
          />
          {
            this.state.selectedSlice &&
            <View style={{
              alignSelf: 'center',
              marginTop: 20,
              justifyContent: 'center',
              width: "85%",
              height: 200,
              backgroundColor: 'white',
              shadowOpacity: 0.5,
              shadowRadius: 5,
              shadowColor: this.state.selectedSlice.color,
              shadowOffset: {height: 0, width: 0},
            }}>
              <View style={{alignItems: 'center'}}>
                <Text style={{fontFamily: POPPINS, fontSize: 20, color: this.state.selectedSlice.color}}>RATING</Text>
                <Text style={{fontFamily: POPPINS, fontSize: 30}}>{this.state.selectedSlice.key}</Text>
              </View>

              <View style={{alignItems: 'center'}}>
                <Text style={{fontFamily: POPPINS, fontSize: 20, color: this.state.selectedSlice.color}}>HOW MANY
                  TIMES</Text>
                <Text style={{
                  fontFamily: POPPINS,
                  fontSize: 30
                }}>{this.state.selectedSlice.value}</Text>
              </View>
            </View>
          }
        </View>
      </View>
    )
  }
}

export default ByRatingDetails