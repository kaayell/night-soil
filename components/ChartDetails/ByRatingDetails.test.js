import {shallow} from "enzyme";
import ByRatingDetails from "./ByRatingDetails";
import React from "react";
import Firebase from "../Firebase/Firebase";

describe('ByRatingDetails', () => {
  it('should render', () => {
    Firebase.getPoopsRef = jest.fn(() => {
      return {
        on: jest.fn(() => Promise.resolve(
          [
            {
              val: () => {
                return "hi"
              }
            },
          ]
        ))
      }
    })
    const wrapper = shallow(<ByRatingDetails />)
    expect(wrapper).toMatchSnapshot()
  });
});