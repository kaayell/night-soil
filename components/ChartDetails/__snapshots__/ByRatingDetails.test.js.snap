// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ByRatingDetails should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "flex": 1,
      "height": "100%",
      "paddingTop": 20,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": "5%",
        "justifyContent": "flex-end",
        "paddingRight": 15,
      }
    }
  >
    <Icon
      color="#9AC0CD"
      name="close"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={24}
      underlayColor="white"
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <PieChart
      data={Array []}
      height={100}
      innerRadius={10}
      outerRadius="95%"
      padAngle={0.05}
      renderDecorator={[Function]}
      sort={[Function]}
      spacing={0}
      style={
        Object {
          "height": 300,
        }
      }
      valueAccessor={[Function]}
      width={100}
    />
  </View>
</View>
`;
